### DISCUSSION TITLE
Search

### DISCUSSION THREAD

**Discussion by anishsarkar** [filamentphp/filament#980] [Help] [answered]
If in the table we use a relation,  is there specific way to do it so that it is searchable? 

I have a table of transaction requests. All statuses of the requests are being kept in a separate table, called statuses. I have two relations on the request table for statuses : one gives all the requests (one to many) for a request but the other one is  Has One Of Many,  which gives me the latest status for a request, using standard laravel. I call second relation in request model as current_status. 

```
public function current_status () : HasOne
{
      return $this->hasOne(RequestStatus::class)->latestOfMany();
}
```

If I show the table column ('current_status.message')   as follows: 
```
TextColumn::make('current_status.message')
                    ->label('Response')
                    ->searchable()
```
The table show the value correctly, but search fails. 

Any help is greatly appreciated. Thanks in advance.

**Comment by danharrin** [comment, marked as answer]
How would you do this search in normal Eloquent?

**Reply by anishsarkar** [reply]
I guess 

``` 
$str = 'message';

PaymentRequest::whereRelation( 
      'current_status', 
      'message', 
      'like', 
      '%' . $str . '%'
)->get();

```

**Reply by danharrin** [reply]
Have you tested that? I think we do that already. What's the error you're getting in Filament?

**Comment by anishsarkar** [comment]
No error, just getting blank table. Probably I typed in something which is not in the database. I will try again and confirm, whether  it was my mistake.

**Reply by anishsarkar** [reply]
My mistake, confused 1 with I. Sorry about that.
